app/src/main/java/com/example/egida/presentation/ui/FitFragment.kt

```diff
@@ -22,63 +26,77 @@ class FitFragment : Fragment() {
 
     private lateinit var binding: FitFragmentBinding
     private lateinit var viewModel: DayViewModel
+    private lateinit var mainViewModel: MainViewModel
 
     override fun onStart() {
         super.onStart()
-        (activity as MainActivity).mAppDrawer.disableDrawer()
+        (activity as MainActivity).closeDrawer()
+//        mainViewModel.closeDrawer()
         Log.d(TAG, " $DAY")
     }
 
     override fun onResume() {
         super.onResume()
-        changeValuesFragmentFields()
-    }
-
-    override fun onStop() {
-        super.onStop()
-        (activity as MainActivity).mAppDrawer.enableDrawer()
-    }
-
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        binding = FitFragmentBinding.inflate(layoutInflater)
-        return binding.root
-    }
-
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel = ViewModelProvider(this).get(DayViewModel::class.java)
 
         binding.minusRunning.setOnClickListener {
-            viewModel.day.running = viewModel.day.running - 1
-            binding.textRunning.text = viewModel.day.running.toString()
+            lifecycleScope.launch {
+                viewModel.day
+                    .collect {
+                        binding.textRunning.text = (it.running--).toString()
+                        binding.textRunning.text = it.running.toString()
```

Comment:
What is the reason to set text twice? Do required operations then set text once.